2025-04-14-January model to send for review.nlogo

```diff
@@ -0,0 +1,1081 @@
+globals
+[
+  pg-B ;; probability of good experimental result (for trying B) if B is better - .5 + advantage-B
+  pg-A ;; probability of good experimental result (for trying B) if A is better - .5 - advantage-B (symmetry here is optional)
+  average-credences ;; a list to store the average credence after each run
+  final-credences ;; A list of lists to store the final credence of each turtle for each run
+  total-runs ;; Specifies the total number of simulation runs desired
+  current-run
+  total-successes
+  total-failures
+  no-convergence
+  percent-success
+  average-p-credence
+  average-p-credences-list
+  running-average-p-credence
+  time-to-finish
+  time-to-finish-list
+  average-time-to-finish
+]
+
+turtles-own [
+  credence ;; credence that B is better
+  last-credence ;; credence B is better before most recent update - there to make it easier to see changes
+  likelihood-positive
+  likelihood-negative
+  odds
+  last-odds
+  my-positive
+  incoming-positive
+  total-positive
+  my-negative
+  incoming-negative
+  total-negative
+  prop-positives
+  prop-negatives
+  sci-positives
+  sci-negatives
+  pos-updates
+  neg-updates
+]
+
+breed [policymakers policymaker]
+breed [scientists scientist]
+breed [propagandists propagandist]
+
+undirected-link-breed [Sci-Scis Sci-Sci]
+undirected-link-breed [Sci-Pols Sci-Pol]
+undirected-link-breed [Prop-Scis Prop-Sci]
+undirected-link-breed [Prop-Pols Prop-Pol]
+
+; THIS BLOCK OF PROCESSES DOES SETUP-Y STUFF
+
+to setup
+  ca
+  ifelse network-type = "cycle" [
+    setup-cycle
+  ] [
+    ifelse network-type = "wheel" [
+      setup-wheel
+    ] [
+      if network-type = "complete" [
+        setup-complete ; Assuming "connected" refers to a fully connected network, and renamed accordingly
+      ]
+    ]
+  ]
+  set average-p-credences-list []
+  set average-credences []
+  set final-credences []
+  set time-to-finish-list []
+  setup-policymakers
+  setup-propagandists
+  set current-run 0
+  set total-successes 0
+  set total-failures 0
+  set total-runs 10000 ;; Example: total runs set to 100, adjust based on your experiment needs
+end
+
+
+; SCIENTIST NETWORK SETUPS
+
+to setup-cycle
+ clear-turtles
+  reset-ticks
+  update-real-probabilities
+  ask patches [ set pcolor blue ]
+  create-scientists number-turtles
+  layout-circle sort scientists 8
+   ; Shift the circle to the right
+  let shift-x max-pxcor / 2 - 14 ; Adjust this to move the circle rightward, closer to the edge
+  ask scientists [
+    setxy xcor + shift-x ycor
+  ]
+  create-cycle-network
+  ask scientists [
+    set shape "circle"
+    set color green
+    initialize-beliefs
+  ]
+end
```

Comment:
It is standard in netlogo to take this approach to setup (sub-procedures for individual setup routines), but consider refactoring everything like `clear-turtles` and `reset-ticks` into the main proc. Not doing so is an easy source of headache in the future.